File: doubly_linked_list/recipe-02/cxx/test_doubly_linked_list.cpp
#include "doubly_linked_list.hpp"

#include <string>

void test() {
    DoublyLinkedList<std::string> dll;
    dll.Append("foo");
    dll.Append("bar");
    dll.Append("biz");
    dll.Append("whew");

    std::cout << "Items in List : " << '\n';
    dll.PrintFoward();

    std::cout << "List after deleting node with data whew" << '\n';
    dll.Delete("whew");
    dll.PrintFoward();

    std::cout << "List count: " << dll.Count() << '\n';
    std::cout << "Print list backwards" << '\n';
    dll.PrintBackward();

    std::cout << "Reverse list " << '\n';
    dll.Reverse();
    dll.PrintFoward();

    std::cout << "Append item to front of list" << '\n';
    dll.InsertHead(std::to_string(55));
    dll.PrintFoward();

    std::cout << "Get First element: " << dll[0] << '\n';
}

int main()
{
    test();
    return 0;
}

File: list/doubly_linked_list/recipe-02/python/test_doubly_linked_list.py
from doubly_linked_list import DoublyLinkedList

def test():
    dll = DoublyLinkedList()
    dll.Append("foo")
    dll.Append("bar")
    dll.Append("biz")
    dll.Append("whew")
    print("Items in List : ")
    dll.PrintFoward()
    print("List after deleting node with data whew")
    dll.Delete("whew")
    dll.PrintFoward()
    
    print("List count: {}".format(dll.count))
    print("Print list backwards")
    dll.PrintBackward()
    
    print("Reverse list ")
    dll.Reverse()
    dll.PrintFoward()
    
    print("Append item to front of list")
    dll.InsertHead(55)
    dll.PrintFoward()
    
    print("Get First element: {}".format(dll[0]))

if __name__ == "__main__":
    test()
